### PR TITLE
Remove all Context.Global* methods

### LINKED ISSUE #385: Remove support for `Global*` methods.
Instead, have the normal flag fetch methods always walk up the context hierarchy until they find their matching flag (or hit the root).

### PR DESCRIPTION
**Pull request by meatballhat**
and change the behavior of the non-Global variants to always search up the
context lineage.

Closes #385

### CODE REVIEW

File: context.go [hunk 1]
```diff
@@ -166,68 +113,63 @@ func (c *Context) Set(name, value string) error {
...
 	return c.parentContext
 }
 
+// Lineage returns *this* context and all of its ancestor contexts in order from
+// child to parent
+func (c *Context) Lineage() []*Context {
+	lineage := []*Context{}
+	cur := c
+
+	for {
```
Comment (jszwedko) [inline context.go]: Feels like you could shrink this up by doing something like:

```
for cur := c; cur != nil; cur = cur.parentContext {
  lineage = append(lineage, cur)
}
```

File: context.go [hunk 2]
```diff
@@ -445,3 +385,12 @@ func normalizeFlags(flags []Flag, set *flag.FlagSet) error {
 	}
 	return nil
 }
+
+func makeFlagNameVisitor(names *[]string) func(*flag.Flag) {
+	return func(f *flag.Flag) {
+		name := strings.Split(f.Name, ",")[0]
+		if name != "help" && name != "version" {
```
Comment (jszwedko) [inline context.go]: Can you elaborate why we don't want to append if the name is `help` or `version`?

**Comment by jszwedko** [comment]
:clap: I'm super happy to see this.

Do you think you could add a couple of tests testing:

* `context.Lineage`
* That the context accessor functions properly traverse up (or maybe just test `lookupFlagSet` since that is where most of the logic is)
* That `IsSet` traverses up

:heart:

File: context.go (hunk 2 above)
Comment (meatballhat) [inline context.go]: I dunno!  This behavior came along for the ride :smile_cat: https://github.com/codegangsta/cli/blob/3a52a3b8124e194986964c547054f0a7b8f56892/context.go#L202-L224

Comment (meatballhat) [inline context.go]: Taking these exclusions out breaks no tests, so I'm taking them out until we can find a good reason to have them :smiley_cat:

File: context.go [hunk 3]
```diff
@@ -13,12 +13,11 @@ import (
 // can be used to retrieve context-specific Args and
 // parsed command-line options.
 type Context struct {
-	App            *App
-	Command        Command
-	flagSet        *flag.FlagSet
-	setFlags       map[string]bool
-	globalSetFlags map[string]bool
-	parentContext  *Context
+	App     *App
```
Comment (jszwedko) [inline context.go]: :clap: This reduced signature feels much better.

File: context.go [hunk 4]
```diff
@@ -138,33 +137,19 @@ func (c *Context) LocalFlagNames() []string {
...
-}
-
 // Lineage returns *this* context and all of its ancestor contexts in order from
 // child to parent
 func (c *Context) Lineage() []*Context {
 	lineage := []*Context{}
-	cur := c
 
-	for {
+	for cur := c; cur != nil; cur = cur.parentContext {
```
Comment (jszwedko) [inline context.go]: :heart_eyes:

File: context_test.go [hunk 5]
```diff
@@ -84,18 +84,22 @@ func TestContext_NArg(t *testing.T) {
...
+	parentCtx := NewContext(nil, parentSet, nil)
+	ctx := NewContext(nil, set, parentCtx)
+
+	set.Parse([]string{"--one-flag", "--two-flag", "--three-flag", "frob"})
+	parentSet.Parse([]string{"--top-flag"})
+
+	expect(t, ctx.IsSet("one-flag"), true)
+	expect(t, ctx.IsSet("two-flag"), true)
+	expect(t, ctx.IsSet("three-flag"), true)
+	expect(t, ctx.IsSet("top-flag"), true)
```
Comment (jszwedko) [inline context_test.go]: :+1:

**Comment by jszwedko** [comment]
Beautiful.

**Pull request merged**